src/guides/v2.3/migration/extend-the-tool.md

```diff
@@ -121,7 +152,245 @@ Steps can include four types of classes:
+                $sourceRecord = $this->recordFactory->create(
+                    ['document' => $sourceDocument, 'data' => $item]
+                );
+                $destinationRecord = $this->recordFactory->create(['document' => $destinationDocument]);
+                $recordTransformer->transform($sourceRecord, $destinationRecord);
+                $recordsToSave->addRecord($destinationRecord);
+            }
+            $this->destination->saveRecords($destinationDocName, $recordsToSave);
+            
+            $tags = $this->getTags($items);
+            $this->destination->saveRecords('greatblog_post_tags', $tags);
+            $this->progress->advance();
+        }
+
+        $this->progress->finish();
+        return true;
+    }
+    ...
+}    
+```
+
+In a Volume class we check if data have been fully migrated `Vendor\Migration\Step\GreatBlog\Volume`:
+
+```php
+class Volume extends AbstractVolume
+{
+    ...
+    /**
+     * @inheritdoc
+     */
+    public function perform()
+    {
+        $documentName = 'greatblog_post';
+        $sourceCount = $this->source->getRecordsCount($documentName);
+        $destinationCount = $this->destination->getRecordsCount($documentName);
+        if ($sourceCount != $destinationCount) {
+            $this->errors[] = sprintf(
+                'Mismatch of entities in the document: %s Source: %s Destination: %s',
+                $documentName,
+                $sourceCount,
+                $destinationCount
+            );
+        }
+
+        return $this->checkForErrors(Logger::ERROR);
+    }
+    ...
+}
+```
+
+To add delta migration functionality you also need to add a new group to `deltalog.xml`. 
+In group specify name of tables that will be checked for changes:
+
+```xml
+<groups>
+    ...
+    <group name="delta_greatblog">
+        <document key="post_id">greatblog_post</document>
+    </group>
+</groups>
+```
+
+Then create `Delta` class `Vendor\Migration\Step\GreatBlog\Delta` that extends `Migration\App\Step\AbstractDelta`:
+
+```php
+class Delta extends AbstractDelta
+{
+    /**
+     * @var string
+     */
+    protected $mapConfigOption = 'greatblog_map_file';
+
+    /**
+     * @var string
+     */
+    protected $groupName = 'delta_greatblog';
+
+    /**
+     * @inheritDoc
+     */
+    public function perform()
+    {
+        $sourceDocumentName = 'greatblog_post';
+        $idKeys = ['post_id'];
+        $page = 0;
+        while (!empty($items = $this->source->getChangedRecords($sourceDocumentName, $idKeys, $page++))) {
+            $this->destination->deleteRecords(
+                'greatblog_post_tags',
+                $idKeys,
+                $items
+            );
+
+            $tags = $this->getTags($items);
+            $this->destination->saveRecords('greatblog_post_tags', $tags);
+        }
+
+        //parent class will take care of greatblog_post records automatically
+        return parent::perform();
+    }
+}
+```
```

Comment:
May I ask you to provide some closing summary here. Like

> After the custom step implementation provided in the examples, the system will take data from the single m1 table, process it using `...` and store the data in `...` 

Thank you!

Reply:
Changes have been made. Thank you